Client/Application/Delegates/AppDelegate.swift

```diff
@@ -344,6 +305,58 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
       
         return shouldPerformAdditionalDelegateHandling
     }
+    
+    func handleReferralLookup(_ urp: UserReferralProgram, checkClipboard: Bool) {
+        let initialOnboarding =
+            Preferences.General.basicOnboardingProgress.value == OnboardingProgress.none.rawValue
+        
+        if initialOnboarding && UIPasteboard.general.hasStrings {
+            log.debug("Skipping URP call at app launch, this is handled in privacy consent onboarding screen")
+            return
+        }
```

Comment:
Im not sure, why is this required? Cant we just only call it after onboarding? Anyone who has already onboarded should have already had URP checked, right?

Reply:
Not sure if I understand, but this check is for 2 cases:
- First launch, no ref code in clipboard, then we don't show the consent screen and URP logic is called from here
- First launch, ref code in clipboard, user kills the app while on the consent screen, on second launch we still wait with urp code until user goes past the privacy consent onboarding

Reply:
Ohhh gotcha, okay that makes sense thanks